src/libs/UserAvatarUtils.ts

```diff
@@ -0,0 +1,327 @@
+        const intVal = Number.parseInt(md5(accountEmail).substring(0, 4), 16);
+        accountIDHashBucket = ((intVal % CONST.DEFAULT_AVATAR_COUNT) + 1) as AvatarRange;
+    } else if (accountID > 0) {
+        accountIDHashBucket = ((accountID % CONST.DEFAULT_AVATAR_COUNT) + 1) as AvatarRange;
+    }
+    return accountIDHashBucket;
+}
+
+/**
+ * Returns the default avatar asset associated with the given accountID.
+ * Special accounts (Concierge, Notifications) have dedicated avatars.
+ * Other accounts get an avatar from the default avatar set based on their accountID hash bucket.
+ *
+ * @param accountID - The user's account ID (defaults to CONST.DEFAULT_NUMBER_ID)
+ * @param accountEmail - The user's email address (for consistency with backend logic)
+ * @param avatarURL - Existing avatar URL to extract avatar number from
+ * @returns The avatar icon asset (SVG component), or undefined if no default avatar matches
+ */
+function getDefaultAvatar({accountID = CONST.DEFAULT_NUMBER_ID, accountEmail, avatarURL}: DefaultAvatarArgsType): IconAsset | undefined {
+    if (accountID === CONST.ACCOUNT_ID.CONCIERGE) {
+        return ConciergeAvatar;
+    }
+    if (accountID === CONST.ACCOUNT_ID.NOTIFICATIONS) {
+        return NotificationsAvatar;
+    }
+
+    return getAvatarLocal(getDefaultAvatarName(accountID, accountEmail, avatarURL));
+}
+
+/**
+ * Returns the custom avatar name (e.g., "default-avatar_5") associated with an account.
+ * This name corresponds to assets in the CustomAvatarCatalog.
+ *
+ * @param accountID - The user's account ID (defaults to CONST.DEFAULT_NUMBER_ID)
+ * @param accountEmail - The user's email address
+ * @param avatarURL - Existing avatar URL
+ * @returns The custom avatar name identifier (e.g., "default-avatar_5")
+ */
+function getDefaultAvatarName(accountID: number = CONST.DEFAULT_NUMBER_ID, accountEmail?: string, avatarURL?: string): DefaultAvatarIDs {
+    return `${DEFAULT_AVATAR_PREFIX}_${getAccountIDHashBucket({accountID, accountEmail, avatarURL})}`;
+}
+
+/**
+ * Returns the CloudFront CDN URL for a custom/default avatar.
+ * This function is used when you need the actual URL string (e.g., for sharing, external display).
+ * Concierge account gets a special hardcoded URL.
+ *
+ * @param accountID - The user's account ID (defaults to CONST.DEFAULT_NUMBER_ID)
+ * @param accountEmail - The user's email address (takes precedence over accountID)
+ * @param avatarURL - Existing avatar URL to extract avatar number from
+ * @returns The CloudFront CDN URL for the avatar image
+ *
+ * @example
+ * // Get avatar URL by account ID
+ * getCustomAvatarURL(12345)
+ * // => 'https://d2k5nsl2zxldvw.cloudfront.net/images/avatars/default-avatar_18.png'
+ *
+ * @example
+ * // Get avatar URL for Concierge
+ * getCustomAvatarURL(CONST.ACCOUNT_ID.CONCIERGE)
+ * // => CONST.CONCIERGE_ICON_URL
+ */
+function getCustomAvatarURL(accountID: number = CONST.DEFAULT_NUMBER_ID, accountEmail?: string, avatarURL?: string): string {
```

Comment:
Should we be changing this from getDefault? I wonder if instead we should separate the generator from the getter. The old `getDefaultAvatarURL` seems to mostly have been used for optimistic generation.

WDYT?